// @pulumi/cloudflare/customHostname.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomHostname = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a Cloudflare custom hostname (also known as SSL for SaaS) resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const example = new cloudflare.CustomHostname("example", {
 *     zoneId: "0da42c8d2132a9ddaf714f9e7c920711",
 *     hostname: "hostname.example.com",
 *     ssls: [{
 *         method: "txt",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/customHostname:CustomHostname example 1d5fdc9e88c8a8c4518b068cd94331fe/0d89c70d-ad9f-4843-b99f-6cc0252067e9
 * ```
 */
class CustomHostname extends pulumi.CustomResource {
    /**
     * Get an existing CustomHostname resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomHostname(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomHostname.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomHostname.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["customMetadata"] = state ? state.customMetadata : undefined;
            resourceInputs["customOriginServer"] = state ? state.customOriginServer : undefined;
            resourceInputs["customOriginSni"] = state ? state.customOriginSni : undefined;
            resourceInputs["hostname"] = state ? state.hostname : undefined;
            resourceInputs["ownershipVerification"] = state ? state.ownershipVerification : undefined;
            resourceInputs["ownershipVerificationHttp"] = state ? state.ownershipVerificationHttp : undefined;
            resourceInputs["ssls"] = state ? state.ssls : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["waitForSslPendingValidation"] = state ? state.waitForSslPendingValidation : undefined;
            resourceInputs["zoneId"] = state ? state.zoneId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.hostname === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostname'");
            }
            if ((!args || args.zoneId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["customMetadata"] = args ? args.customMetadata : undefined;
            resourceInputs["customOriginServer"] = args ? args.customOriginServer : undefined;
            resourceInputs["customOriginSni"] = args ? args.customOriginSni : undefined;
            resourceInputs["hostname"] = args ? args.hostname : undefined;
            resourceInputs["ssls"] = args ? args.ssls : undefined;
            resourceInputs["waitForSslPendingValidation"] = args ? args.waitForSslPendingValidation : undefined;
            resourceInputs["zoneId"] = args ? args.zoneId : undefined;
            resourceInputs["ownershipVerification"] = undefined /*out*/;
            resourceInputs["ownershipVerificationHttp"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomHostname.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomHostname = CustomHostname;
/** @internal */
CustomHostname.__pulumiType = 'cloudflare:index/customHostname:CustomHostname';
