// @pulumi/cloudflare/ipsecTunnel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpsecTunnel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a resource, that manages IPsec tunnels for Magic Transit.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const example = new cloudflare.IpsecTunnel("example", {
 *     accountId: "f037e56e89293a057740de681ac9abbe",
 *     name: "IPsec_1",
 *     customerEndpoint: "203.0.113.1",
 *     cloudflareEndpoint: "203.0.113.1",
 *     interfaceAddress: "192.0.2.0/31",
 *     description: "Tunnel for ISP X",
 *     healthCheckEnabled: true,
 *     healthCheckTarget: "203.0.113.1",
 *     healthCheckType: "reply",
 *     psk: "asdf12341234",
 *     allowNullCipher: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/ipsecTunnel:IpsecTunnel example <account_id>/<tunnel_id>
 * ```
 */
class IpsecTunnel extends pulumi.CustomResource {
    /**
     * Get an existing IpsecTunnel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IpsecTunnel(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IpsecTunnel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IpsecTunnel.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["allowNullCipher"] = state ? state.allowNullCipher : undefined;
            resourceInputs["cloudflareEndpoint"] = state ? state.cloudflareEndpoint : undefined;
            resourceInputs["customerEndpoint"] = state ? state.customerEndpoint : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["fqdnId"] = state ? state.fqdnId : undefined;
            resourceInputs["healthCheckDirection"] = state ? state.healthCheckDirection : undefined;
            resourceInputs["healthCheckEnabled"] = state ? state.healthCheckEnabled : undefined;
            resourceInputs["healthCheckRate"] = state ? state.healthCheckRate : undefined;
            resourceInputs["healthCheckTarget"] = state ? state.healthCheckTarget : undefined;
            resourceInputs["healthCheckType"] = state ? state.healthCheckType : undefined;
            resourceInputs["hexId"] = state ? state.hexId : undefined;
            resourceInputs["interfaceAddress"] = state ? state.interfaceAddress : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["psk"] = state ? state.psk : undefined;
            resourceInputs["remoteId"] = state ? state.remoteId : undefined;
            resourceInputs["replayProtection"] = state ? state.replayProtection : undefined;
            resourceInputs["userId"] = state ? state.userId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.cloudflareEndpoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cloudflareEndpoint'");
            }
            if ((!args || args.customerEndpoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'customerEndpoint'");
            }
            if ((!args || args.interfaceAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'interfaceAddress'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["allowNullCipher"] = args ? args.allowNullCipher : undefined;
            resourceInputs["cloudflareEndpoint"] = args ? args.cloudflareEndpoint : undefined;
            resourceInputs["customerEndpoint"] = args ? args.customerEndpoint : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["fqdnId"] = args ? args.fqdnId : undefined;
            resourceInputs["healthCheckDirection"] = args ? args.healthCheckDirection : undefined;
            resourceInputs["healthCheckEnabled"] = args ? args.healthCheckEnabled : undefined;
            resourceInputs["healthCheckRate"] = args ? args.healthCheckRate : undefined;
            resourceInputs["healthCheckTarget"] = args ? args.healthCheckTarget : undefined;
            resourceInputs["healthCheckType"] = args ? args.healthCheckType : undefined;
            resourceInputs["hexId"] = args ? args.hexId : undefined;
            resourceInputs["interfaceAddress"] = args ? args.interfaceAddress : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["psk"] = (args === null || args === void 0 ? void 0 : args.psk) ? pulumi.secret(args.psk) : undefined;
            resourceInputs["remoteId"] = args ? args.remoteId : undefined;
            resourceInputs["replayProtection"] = args ? args.replayProtection : undefined;
            resourceInputs["userId"] = args ? args.userId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["psk"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(IpsecTunnel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IpsecTunnel = IpsecTunnel;
/** @internal */
IpsecTunnel.__pulumiType = 'cloudflare:index/ipsecTunnel:IpsecTunnel';
