// @pulumi/cloudflare/provider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The provider type for the cloudflare package. By default, resources use package-wide configuration
 * settings, however an explicit `Provider` instance may be created and passed during resource
 * construction to achieve fine-grained programmatic control over provider settings. See the
 * [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.
 */
class Provider extends pulumi.ProviderResource {
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === "pulumi:providers:" + Provider.__pulumiType;
    }
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e;
        let resourceInputs = {};
        opts = opts || {};
        {
            resourceInputs["apiBasePath"] = args ? args.apiBasePath : undefined;
            resourceInputs["apiClientLogging"] = pulumi.output((_a = (args ? args.apiClientLogging : undefined)) !== null && _a !== void 0 ? _a : (utilities.getEnvBoolean("CLOUDFLARE_API_CLIENT_LOGGING") || false)).apply(JSON.stringify);
            resourceInputs["apiHostname"] = args ? args.apiHostname : undefined;
            resourceInputs["apiKey"] = args ? args.apiKey : undefined;
            resourceInputs["apiToken"] = args ? args.apiToken : undefined;
            resourceInputs["apiUserServiceKey"] = args ? args.apiUserServiceKey : undefined;
            resourceInputs["email"] = args ? args.email : undefined;
            resourceInputs["maxBackoff"] = pulumi.output((_b = (args ? args.maxBackoff : undefined)) !== null && _b !== void 0 ? _b : (utilities.getEnvNumber("CLOUDFLARE_MAX_BACKOFF") || 30)).apply(JSON.stringify);
            resourceInputs["minBackoff"] = pulumi.output((_c = (args ? args.minBackoff : undefined)) !== null && _c !== void 0 ? _c : (utilities.getEnvNumber("CLOUDFLARE_MIN_BACKOFF") || 1)).apply(JSON.stringify);
            resourceInputs["retries"] = pulumi.output((_d = (args ? args.retries : undefined)) !== null && _d !== void 0 ? _d : (utilities.getEnvNumber("CLOUDFLARE_RETRIES") || 3)).apply(JSON.stringify);
            resourceInputs["rps"] = pulumi.output((_e = (args ? args.rps : undefined)) !== null && _e !== void 0 ? _e : (utilities.getEnvNumber("CLOUDFLARE_RPS") || 4)).apply(JSON.stringify);
            resourceInputs["userAgentOperatorSuffix"] = args ? args.userAgentOperatorSuffix : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'cloudflare';
