// @pulumi/cloudflare/regionalHostname.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionalHostname = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a Data Localization Suite Regional Hostname.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * // Regionalized hostname record resources are managed independently from the
 * // Regionalized Hostname resources.
 * const example = new cloudflare.Record("example", {
 *     zoneId: "0da42c8d2132a9ddaf714f9e7c920711",
 *     name: "example.com",
 *     content: "192.0.2.1",
 *     type: "A",
 *     ttl: 3600,
 * });
 * // The cloudflare_regional_hostname resource may exist with or without its
 * // corresponding record resource.
 * const exampleRegionalHostname = new cloudflare.RegionalHostname("example", {
 *     zoneId: "0da42c8d2132a9ddaf714f9e7c920711",
 *     hostname: "example.com",
 *     regionKey: "eu",
 * });
 * ```
 */
class RegionalHostname extends pulumi.CustomResource {
    /**
     * Get an existing RegionalHostname resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegionalHostname(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegionalHostname.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegionalHostname.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdOn"] = state ? state.createdOn : undefined;
            resourceInputs["hostname"] = state ? state.hostname : undefined;
            resourceInputs["regionKey"] = state ? state.regionKey : undefined;
            resourceInputs["zoneId"] = state ? state.zoneId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.hostname === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostname'");
            }
            if ((!args || args.regionKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'regionKey'");
            }
            if ((!args || args.zoneId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["hostname"] = args ? args.hostname : undefined;
            resourceInputs["regionKey"] = args ? args.regionKey : undefined;
            resourceInputs["zoneId"] = args ? args.zoneId : undefined;
            resourceInputs["createdOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegionalHostname.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RegionalHostname = RegionalHostname;
/** @internal */
RegionalHostname.__pulumiType = 'cloudflare:index/regionalHostname:RegionalHostname';
