// @pulumi/cloudflare/webAnalyticsRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAnalyticsRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Provides a Cloudflare Web Analytics Rule resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const example = new cloudflare.WebAnalyticsSite("example", {
 *     accountId: "f037e56e89293a057740de681ac9abbe",
 *     zoneTag: "0da42c8d2132a9ddaf714f9e7c920711",
 *     autoInstall: true,
 * });
 * const exampleWebAnalyticsRule = new cloudflare.WebAnalyticsRule("example", {
 *     accountId: "f037e56e89293a057740de681ac9abbe",
 *     rulesetId: example.rulesetId,
 *     host: "*",
 *     paths: ["/excluded"],
 *     inclusive: false,
 *     isPaused: false,
 * }, {
 *     dependsOn: [example],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/webAnalyticsRule:WebAnalyticsRule example <account_id>/<ruleset_id>/<rule_id>
 * ```
 */
class WebAnalyticsRule extends pulumi.CustomResource {
    /**
     * Get an existing WebAnalyticsRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WebAnalyticsRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAnalyticsRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAnalyticsRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["host"] = state ? state.host : undefined;
            resourceInputs["inclusive"] = state ? state.inclusive : undefined;
            resourceInputs["isPaused"] = state ? state.isPaused : undefined;
            resourceInputs["paths"] = state ? state.paths : undefined;
            resourceInputs["rulesetId"] = state ? state.rulesetId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.host === undefined) && !opts.urn) {
                throw new Error("Missing required property 'host'");
            }
            if ((!args || args.inclusive === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inclusive'");
            }
            if ((!args || args.isPaused === undefined) && !opts.urn) {
                throw new Error("Missing required property 'isPaused'");
            }
            if ((!args || args.paths === undefined) && !opts.urn) {
                throw new Error("Missing required property 'paths'");
            }
            if ((!args || args.rulesetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rulesetId'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["host"] = args ? args.host : undefined;
            resourceInputs["inclusive"] = args ? args.inclusive : undefined;
            resourceInputs["isPaused"] = args ? args.isPaused : undefined;
            resourceInputs["paths"] = args ? args.paths : undefined;
            resourceInputs["rulesetId"] = args ? args.rulesetId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WebAnalyticsRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WebAnalyticsRule = WebAnalyticsRule;
/** @internal */
WebAnalyticsRule.__pulumiType = 'cloudflare:index/webAnalyticsRule:WebAnalyticsRule';
